axelrod/tournament.py

```diff
@@ -11,21 +11,20 @@
 from axelrod.player import Player
 from .game import Game
 from .match import Match
-from .match_generator import (
-    RoundRobinMatches, ProbEndRoundRobinMatches, SpatialMatches,
-    ProbEndSpatialMatches, MatchGenerator)
+from .match_generator import MatchGenerator
 from .result_set import ResultSetFromFile, ResultSet
 
-from typing import List
+from typing import List, Tuple
 
 
 class Tournament(object):
 
     def __init__(self, players: List[Player],
-                 match_generator: MatchGenerator = RoundRobinMatches,
-                 name: str = 'axelrod', game: Game = None, turns: int = 200,
-                 repetitions: int = 10,
-                 noise: float = 0, with_morality: bool = True) -> None:
+                 match_generator: MatchGenerator = MatchGenerator,
```

Comment:
> Is match_generator still needed as a parameter?

I'd be happy to remove it. The only reason we really have it is to allow for the `making new tournaments` idea: http://axelrod.readthedocs.io/en/stable/tutorials/advanced/making_tournaments.html

But that's such a niche thing I'd be happy to take it out.

> 
> Would it be better for Tournament to take a dictionary of match parameters?

I don't think so. It's essentially a dictionary anyway (just `kwargs`) and as is is more helpful to the user (autocompletion etc...).